CMakeLists.txt

```diff
@@ -244,7 +247,12 @@ message(STATUS "Build Type: ${CMAKE_BUILD_TYPE}")
 
 # Build with C++11 and SSE3 by default
 # TODO(wesm): These compiler warning suppressions should be removed one by one
-SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -msse3 -Wall")
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall")
+
+if (PARQUET_USE_SSE)
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse4.2")
```

Comment:
AFAIR, `-march=native` is more general: it checks what SSE are supported and enables corresponding flags (such as `-msse`, `-msse2`, etc.).
Anyway, if you are interested specifically in SSE4, then `-msse4.2` does the job.